test/external_tests.py

```diff
@@ -0,0 +1,167 @@
+        ret = run_cmd(f"{test_script_path} {solc_binary_type} {solc_binary_path}")
+        if ret != 0:
+            return ret
+    return 0
+
+
+def run_external_tests(args: dict) -> int:
+    solc_binary_type = args.get("solc_binary_type")
+    solc_binary_path = args.get("solc_binary_path")
+
+    all_test_scripts = external_tests_scripts()
+    if args.get("run_all"):
+        return run_test_script(solc_binary_type, solc_binary_path, all_test_scripts)
+    else:
+        selected_tests = args.get("selected_tests")
+        if selected_tests:
+            diff_set = set(selected_tests) - set(all_test_scripts.keys())
+            if diff_set:
+                raise ExternalTestNotFound(
+                    f"External test(s): {' '.join(diff_set)} not found"
+                )
+            return run_test_script(
+                solc_binary_type,
+                solc_binary_path,
+                {k: all_test_scripts[k] for k in selected_tests},
+            )
+    raise ExternalTestNotFound(
+        "External test was not selected. Please use --run or --run-all option"
+    )
+
+
+def parse_commandline() -> Namespace:
+    script_description = "Script to run external Solidity tests."
+
+    parser = ArgumentParser(description=script_description)
+    subparser = parser.add_subparsers()
+    list_command = subparser.add_parser(
+        "list",
+        help="List all available external tests.",
+    )
+    list_command.set_defaults(cmd=display_available_external_tests)
+
+    run_command = subparser.add_parser(
+        "run",
+        help="Run external tests.",
+    )
+    run_command.set_defaults(cmd=run_external_tests)
+
+    run_command.add_argument(
+        "--solc-binary-type",
+        dest="solc_binary_type",
+        type=str,
+        required=True,
+        choices=["native", "solcjs"],
+        help="Type of the solidity compiler binary to be used.",
+    )
+    run_command.add_argument(
+        "--solc-binary-path",
+        dest="solc_binary_path",
+        type=Path,
+        required=True,
+        help="Path to the solidity compiler binary.",
+    )
+
+    running_mode = run_command.add_mutually_exclusive_group()
+    running_mode.add_argument(
+        "--run",
+        metavar="TEST_NAME",
+        dest="selected_tests",
+        nargs="+",
+        default=[],
+        help="Run one or more given external tests.",
+    )
+    running_mode.add_argument(
+        "--run-all",
+        dest="run_all",
+        default=False,
+        action="store_true",
+        help="Run all available external tests.",
+    )
+
+    args = parser.parse_args()
+    if not hasattr(args, "cmd"):
+        parser.print_help()
+        raise OptionNotFound
+    return args
+
+
+def main():
+    try:
+        args = parse_commandline()
+        return args.cmd(vars(args))
+    except OptionNotFound:
+        return 1
+    except ExternalTestNotFound as exception:
+        print(f"Error: {exception}", file=sys.stderr)
+        return 1
```

Comment:
You're returning the same error code (`1`) for different types of failures; even if never end up actually using the error codes (i.e. only checking if it's `0` or not), I still think these should be different.

Reply:
I adjust the return exit codes. If you have any suggestion to do it differently I can change it as well :)
The error codes are supposed to the used/handled by the caller of the script. In our case it will the be CI scripts or your terminal when you call the script directly, e.g.:
```sh
./test/external_tests.py list
# or
./test/external_tests.py test --solc-binary-type native --solc-binary-path $(which solc) --run uniswap
```